### DISCUSSION TITLE
Adding and stretching/distorting an image onto the canvas path

### DISCUSSION THREAD

**Discussion by simasio** [dlemstra/Magick.NET#917] [Help] [answered]
I was working on a small program that bends the image, but I just couldn't seem to find any way to fill the following image to a new shape.
I am coding in .net c# but I just can't find a way to fill in the image, stretching, distorting it to a declared canvas path.
All I could achieve using the default GDI+ library was this:
So I wonder if using ImageMagick library would help me doing such distortion/transformation, so I could achieve my desired result

Thanks in advance!
Have a good day!
- Simas

**Comment by dlemstra** [comment, marked as answer]
You can use the `Distort` method for this:

```C#
using (var image = new MagickImage("input.jpg"))
{
    // Make the new pixels transparent
    image.VirtualPixelMethod = VirtualPixelMethod.Transparent;

    // You probably need to tweek the value below.
    image.Distort(DistortMethod.Arc, 50);

    image.Write("output.jpg");
}
```
